src/lib.rs

```diff
@@ -3358,6 +3361,22 @@ safety_comment! {
     unsafe_impl!(Option<NonZeroIsize>: TryFromBytes, FromZeros, FromBytes, IntoBytes);
 }
 
+safety_comment! {
+    /// SAFETY:
+    /// While it's not fully documented, the consensus is that `Box<T>` does not
+    /// contain any `UnsafeCell`s for `T: Sized` [1].
+    ///
+    /// [1] https://github.com/rust-lang/unsafe-code-guidelines/issues/492
+    ///
+    /// TODO(#896): Write a more complete safety proof before the next stable
+    /// release.
+    #[cfg(feature = "alloc")]
+    unsafe_impl!(
+        #[cfg_attr(doc_cfg, doc(cfg(feature = "alloc")))]
+        T: Sized => NoCell for Box<T>
+    );
+}
```

Comment:
Generally speaking, we need to justify that *none* of the primitive types are `!Freeze`, not just `Box<T>`.

Reply:
I agree, but it's not clear that `Box<T>` *is* actually a primitive type. Obviously it is in practice, but it's not clear that it's guaranteed as such. Also, I'm not sure this is the right location for a TODO comment about documenting other primitive types.